src/base/bittorrent/session.cpp

```diff
@@ -296,6 +317,8 @@ Session::Session(QObject *parent)
     , m_encryption(BITTORRENT_SESSION_KEY("Encryption"), 0)
     , m_isForceProxyEnabled(BITTORRENT_SESSION_KEY("ForceProxy"), true)
     , m_isProxyPeerConnectionsEnabled(BITTORRENT_SESSION_KEY("ProxyPeerConnections"), false)
+    , m_chokingAlgorithm(BITTORRENT_SESSION_KEY("ChokingAlgorithm"), 0, clampValue(0, 2))
```

Comment:
You can't mix interface of components of an application with interface of underlying library. If you allow users only two choices then Session::chockingAlgorithm property has only two possible values independently of the way it really implemented.